Metrics/start-windows-build.ps1

```diff
@@ -0,0 +1,365 @@
+        [Parameter(Mandatory=$true)]
+        [string]$BuildErrorMessage
+    )
+    $stdoutFile = Join-Path $METRICS_BUILD_LOGS_DIR $StdoutFileName
+    $stderrFile = Join-Path $METRICS_BUILD_LOGS_DIR $StderrFileName
+    New-Item -ItemType File -Path $stdoutFile -Force
+    New-Item -ItemType File -Path $stderrFile -Force
+    $logsUrl = Get-BuildLogsUrl
+    $stdoutUrl = "${logsUrl}/${StdoutFileName}"
+    $stderrUrl = "${logsUrl}/${StderrFileName}"
+    $command = $ProcessPath -replace '\\', '\\'
+    if($ArgumentList.Count) {
+        $ArgumentList | Foreach-Object { $command += " $($_ -replace '\\', '\\')" }
+    }
+    try {
+        Wait-ProcessToFinish -ProcessPath $ProcessPath -ArgumentList $ArgumentList `
+                             -StandardOutput $stdoutFile -StandardError $stderrFile
+        $msg = "Successfully executed: $command"
+    } catch {
+        $msg = "Failed command: $command"
+        $global:PARAMETERS["BUILD_STATUS"] = 'FAIL'
+        $global:PARAMETERS["LOGS_URLS"] += $($stdoutUrl, $stderrUrl)
+        $global:PARAMETERS["FAILED_COMMAND"] = $command
+        Write-Output "Exception: $($_.ToString())"
+        Throw $BuildErrorMessage
+    } finally {
+        Write-Output $msg
+        Write-Output "Stdout log available at: $stdoutUrl"
+        Write-Output "Stderr log available at: $stderrUrl"
+    }
+}
+
+function Get-LatestCommitID {
+    if(!$global:LATEST_COMMIT_ID) {
+        Throw "Failed to get the latest dcos-metrics commit ID. Perhaps it has not saved."
+    }
+    return $global:LATEST_COMMIT_ID
+}
+
+function Set-LatestMetricsCommit {
+    Push-Location $METRICS_GIT_REPO_DIR
+    try {
+        if($CommitID) {
+            Start-ExternalCommand { git.exe reset --hard $CommitID } -ErrorMessage "Failed to set Metrics git repo last commit to: $CommitID."
+        }
+        Start-ExternalCommand { git.exe log -n 1 } -ErrorMessage "Failed to get the latest commit message for the Metrics git repo" | Out-File "$METRICS_BUILD_LOGS_DIR\latest-commit.log"
+        $commitId = Start-ExternalCommand { git.exe log --format="%H" -n 1 } -ErrorMessage "Failed to get the latest commit id for the dcos-metrics git repo"
+        Set-Variable -Name "LATEST_COMMIT_ID" -Value $commitId -Scope Global
+    } finally {
+        Pop-Location
+    }
+}
+
+function New-TestingEnvironment {
+    Write-Output "Creating new tests environment"
+    Start-EnvironmentCleanup # Do an environment cleanup to make sure everything is fresh
+    New-Directory $METRICS_DIR
+    New-Directory $METRICS_BUILD_OUT_DIR -RemoveExisting
+    New-Directory $METRICS_BUILD_LOGS_DIR
+    $global:PARAMETERS["BRANCH"] = $Branch
+    Start-GitClone -Path $METRICS_GIT_REPO_DIR -URL $GitURL -Branch $Branch
+    Set-LatestMetricsCommit
+    Start-GitClone -Path $METRICS_DCOS_WINDOWS_GIT_REPO_DIR -URL $DCOS_WINDOWS_GIT_URL
+    Start-ExternalCommand { git.exe config --global user.email "ostcauto@microsoft.com" } -ErrorMessage "Failed to set git user email"
```

Comment:
Please remove these 2 `git.exe config` sets as they won't be required anymore. This is a one time setup to be done when getting the Jenkins Windows build server. I'll be removing these from the other scripts as well.

Reply:
I will remove them